datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -349,7 +349,14 @@ def _normalize_tags_type(self, tags):
 
     def warning(self, warning_message):
         warning_message = str(warning_message)
-        self.log.warning(warning_message)
+
+        frame = inspect.currentframe().f_back
+        lineno = frame.f_lineno
+        filename = frame.f_code.co_filename
```

Comment:
Let's `from os.path import basename`

```suggestion
        filename = basename(frame.f_code.co_filename)
```